nilearn/regions/region_extractor.py

```diff
@@ -169,7 +170,14 @@ def connected_regions(maps_img, min_region_size=1350,
         index_of_each_map.extend([index] * len(regions))
         all_regions_imgs.extend(regions)
 
-    regions_extracted_img = concat_niimgs(all_regions_imgs)
+    if len(all_regions_imgs) == 0 and (label_maps == 0).all():
```

Comment:
Checking `len(all_regions_imgs)` is enough. Plus, checking `(label_maps == 0).all()` is wrong because this is a variable local to the for loop above so its value depends on the last component only.